compiler-cli/src/docs.rs

```diff
@@ -72,16 +78,32 @@ pub fn build() -> Result<()> {
     crate::fs::delete_dir(&out)?;
     crate::fs::write_outputs_under(&outputs, &out)?;
 
+    let index_html = out.join("index.html");
+
     println!(
-        "\nThe documentation for {package} has been rendered to \n{out}/index.html",
+        "\nThe documentation for {package} has been rendered to \n{index_html}",
         package = config.name,
-        out = out.to_string_lossy()
+        index_html = index_html.to_string_lossy()
     );
 
+    if options.open {
+        open_docs(&index_html);
+    }
+
     // We're done!
     Ok(())
 }
 
+/// Opens the indicated path in the default program configured by the system.
+///
+/// For the docs this will generally be a browser (unless some other program is
+/// configured as the default for `.html` files).
+fn open_docs(path: &Path) {
+    if let Err(error) = opener::open(path) {
+        eprintln!("Failed to open docs:\n{}", error);
```

Comment:
If we failed to open the docs, log the error to stderr, but don't fail the operation overall (since the docs themselves were still built successfully).

Reply:
Let's use the same error printing mechanism as the others, for consistency. The file IO errors are probably a good one to look at.